pkg/odo/cli/pipelines/ui/ui.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+
+	return path
+}
+
+func EnterDockercfg() string {
+	var path string
+	var prompt *survey.Input
+	prompt = &survey.Input{
+		Message: "Path to config.json which authenticates image pushes to the desired image registry, the default is <insert default>?",
+		Help:    "The secret present in the file path generates a secure secret that authenticates the push of the image built when the app-ci pipeline is run. The image along with the necessary labels will be present on the upstream image repository of choice.",
+		Default: "~/.docker/config.json",
+	}
+
+	err := survey.AskOne(prompt, &path, nil)
+	ui.HandleError(err)
+
+	return path
+}
+
+func EnterImageRepoExternalRepository() string {
+	var path string
+	var prompt *survey.Input
+	prompt = &survey.Input{
+		Message: "Image repository of the form <registry>/<username>/<repository> which is used to push newly built images.",
+		Help:    "By default images are built from source, whenever there is a push to the repository for your service source code and this image will be pushed to the image repository specified in this parameter, if the value is of the form <registry>/<username>/<repository>, then it assumed that it is an upstream image repository e.g. Quay, if its of the form <project>/<app> the internal registry present on the current cluster will be used as the image repository.",
+	}
+
+	err := survey.AskOne(prompt, &path, survey.Required)
+	ui.HandleError(err)
+
+	return path
+}
+
+func EnterOutputPath() string {
+	var path string
+	var prompt *survey.Input
+	prompt = &survey.Input{
+		Message: "Provide a path to write GitOps resources?",
+		Help:    "This is the path where the GitOps repository configuration is stored locally before you push it to the repository GitopsRepoURL <fill in the string from GitOpsRepoURL>",
+		Default: ".",
+	}
+
+	err := survey.AskOne(prompt, &path, nil)
+	ui.HandleError(err)
+
+	return path
+}
+
+func EnterGitWebhookSecret() string {
+	var path string
+	var prompt *survey.Input
+	prompt = &survey.Input{
+		Message: "Provide a secret that we can use to authenticate incoming hooks from your Git hosting service for the GitOps repository. (if not provided, it will be auto-generated)",
+		Help:    "The webhook secret is a secure string you plan to use to authenticate pull/push requests to the version control system of your choice, this secure string will be added to the webhook sealed secret created to enhance security. Choose a secure string of your choice for this field.",
+	}
+
+	err := survey.AskOne(prompt, &path, nil)
+	ui.HandleError(err)
+
+	return path
+}
+
+func EnterSealedSecretService() string {
+	var path string
+	var prompt *survey.Input
+	prompt = &survey.Input{
+		Message: "Name of the Sealed Secrets Services that encrypts secrets <sealed-secrets-controller>",
```

Comment:
default is already added. We don't have to mention it in our questions

Reply:
This query has been addressed too.